staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -337,6 +337,7 @@ type PersistentVolumeSpec struct {
 	// claim.VolumeName is the authoritative bind between PV and PVC.
 	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#binding
 	// +optional
+	// +structType=granular
```

Comment:
I'm not sure our SSA code honors this annotation on a field... based on your experiments, it seems like it doesn't?

Reply:
Correct, SSA does not honor it. I only pushed the code to share the diff/experiement while we decide how to move forward. To me our choices seem to be to either:

1.) Drop `atomic` from ObjectReference, make SSA honor field-level annotations and attach `atomic` as a field-level reference where necessary.
2.) Give `PersistentVolume` its own `ObjectReference` type

Reply:
1 seems best, though its unfortunate it requires SSA changes

2 and 3 leave ObjectReference in a non-ideal state for other out of tree consumers

3 is too disruptive for a patch backport (breaks go compatibility of API types)

Reply:
I am in favor of #1 in two parts:

a. Drop `atomic` from `ObjectReference`
b. Update `SSA` to allow field-level annotations of `structType`

Reply:
Agreed, we need to stop the bleeding with a., we can do cosmetic changes with b. as needed later.

Reply:
if we change a and backport it, we've also modified all the other API uses of this type... have we considered whether *those* changes will have negative consequences?

Reply:
I'm not aware of anything client-wise (especially in client-go) using the structType to change the generation of anything. I'm not aware of anything client-side that uses this specifically to make any determination outside possibly of calling the Update/Apply function of the FieldManager. That includes kubectl who never uses this to do anything relative to Strategic-Merge-Patch or anything else.

The FieldManager has some logic to deal with such changes in schema (we have a test in the smd repo for these, please we now have a test in this PR to verify that it works with actual types). The main consequence of this change is that we'll have a dangling ownership of the struct which isn't going to be honored. It'll probably stay "forever" without expected impact.